### PR TITLE
Allow creating a admin `ClusterRoleBinding`

### PR DESCRIPTION
**Pull request by tigrato**
This PR adds the possibility of creating a cluster role binding between a group whose name is defined by `adminClusterRolebinding.name` and the built-in `cluster-admin` `ClusterRole`. This is particularly useful for GKE Autopilot clusters where it's not possible to use the default `system:masters` group because authz Warden security module prevents impersonating system-wide identities.

When the chart detects that the target cluster is a GKE cluster - `version: v1.x.x-gke.<build>` - it will automatically create the `ClusterRoleBinding` and print a warning message with the following payload:

```
NOTES:
WARNING: GKE Autopilot clusters forbid users from impersonating system-wide identities.
This means you won't be able to use the `system:masters` Kubernetes Group in
the Teleport Roles for GKE Autopilot clusters.

Given that you installed Teleport on a GKE cluster, we recommend you use the
Kubernetes Group `cluster-admin` instead of `system:masters` in the Teleport Roles
for GKE Autopilot clusters.

To do so, you can use the following Teleport Role resource:

 kind: role
 metadata:
   name: gke-kube-access
 version: v7
 spec:
   allow:
     kubernetes_labels:
       '*': '*'
     kubernetes_groups:
     - "cluster-admin"


This chart automatically created the `cluster-admin` Kubernetes Group for you and
assigned it admin privileges on the Kubernetes cluster.

Consult the built-in security features that GKE Autopilot enforces:
https://cloud.google.com/kubernetes-engine/docs/concepts/autopilot-security#built-in-security
```

Part of #28506

### CODE REVIEW

**Review by strideynet — APPROVED** [review]

File: examples/chart/teleport-kube-agent/values.yaml [hunk 1]
```diff
@@ -201,6 +201,15 @@ storage:
   storageClassName: ""
   requests: 128Mi
 
+# Settings for configuring an cluster admin role binding.
+# This is useful for granting cluster admin permissions to a Kubernetes Group
+# other than the default "system:masters" group.
+# GKE Autopilot clusters forbid using the "system:masters" group for impersonation
+# and require a custom group to be used instead.
+adminClusterRolebinding:
```
Comment (webvictim) [inline examples/chart/teleport-kube-agent/values.yaml:209]:
```suggestion
adminClusterRoleBinding:
```

File: examples/chart/teleport-kube-agent/tests/admin_clusterrolebinding_test.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+suite: AdminClusterRoleBinding
+templates:
+  - admin_clusterrolebinding.yaml
+tests:
+  - it: don't generate a admin cluster role binding when adminClusterRolebinding.create is false
```
Comment (webvictim) [inline examples/chart/teleport-kube-agent/tests/admin_clusterrolebinding_test.yaml:5]:
```suggestion
  - it: don't generate a admin cluster role binding when adminClusterRoleBinding.create is false
```

File: examples/chart/teleport-kube-agent/tests/admin_clusterrolebinding_test.yaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    asserts:
+      - hasDocuments:
+          count: 0
+  - it: generate a admin cluster role binding when adminClusterRolebinding.create is true
```
Comment (webvictim) [inline examples/chart/teleport-kube-agent/tests/admin_clusterrolebinding_test.yaml:9]:
```suggestion
  - it: generate a admin cluster role binding when adminClusterRoleBinding.create is true
```

File: examples/chart/teleport-kube-agent/tests/admin_clusterrolebinding_test.yaml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    set:
+      adminClusterRolebinding:
```
Comment (webvictim) [inline examples/chart/teleport-kube-agent/tests/admin_clusterrolebinding_test.yaml:11]:
```suggestion
      adminClusterRoleBinding:
```

File: examples/chart/teleport-kube-agent/tests/admin_clusterrolebinding_test.yaml [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    asserts:
+      - hasDocuments:
+          count: 1
+      - isKind:
+          of: ClusterRoleBinding
+      - equal:
+          path: subjects[0].name
+          value: cluster-admin
+      - matchSnapshot: {}
+  - it: generate a admin cluster role binding when adminClusterRolebinding.create is true and adminClusterRolebinding.name is set
```
Comment (webvictim) [inline examples/chart/teleport-kube-agent/tests/admin_clusterrolebinding_test.yaml:22]:
```suggestion
  - it: generate a admin cluster role binding when adminClusterRoleBinding.create is true and adminClusterRoleBinding.name is set
```

File: examples/chart/teleport-kube-agent/tests/admin_clusterrolebinding_test.yaml [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    set:
+      adminClusterRolebinding:
```
Comment (webvictim) [inline examples/chart/teleport-kube-agent/tests/admin_clusterrolebinding_test.yaml:24]:
```suggestion
      adminClusterRoleBinding:
```

File: examples/chart/teleport-kube-agent/templates/admin_clusterrolebinding.yaml [hunk 7]
```diff
@@ -0,0 +1,24 @@
+{{/* GKE Autopilot clusters forbid users from impersonating system:masters
+Groups. This is a security measure released under the GKE Warden authz module
+https://cloud.google.com/kubernetes-engine/docs/concepts/autopilot-security#built-in-security
+Because of this limitation, users are unable to specify kubernetes_groups=["system:masters"]
+in Teleport, so we create a Kubernetes Group called cluster-admin when we detect
+that the underlying cluster is a GKE cluster.  */}}
+{{- if or (contains "-gke." .Capabilities.KubeVersion.Version) (.Values.adminClusterRolebinding.create) -}}
```
Comment (webvictim) [inline examples/chart/teleport-kube-agent/templates/admin_clusterrolebinding.yaml:7]:
```suggestion
{{- if or (contains "-gke." .Capabilities.KubeVersion.Version) (.Values.adminClusterRoleBinding.create) -}}
```

File: examples/chart/teleport-kube-agent/templates/admin_clusterrolebinding.yaml [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+{{- $groupName := (coalesce .Values.adminClusterRolebinding.name "cluster-admin") }}
```
Comment (webvictim) [inline examples/chart/teleport-kube-agent/templates/admin_clusterrolebinding.yaml:8]:
```suggestion
{{- $groupName := (coalesce .Values.adminClusterRoleBinding.name "cluster-admin") }}
```

File: examples/chart/teleport-kube-agent/templates/NOTES.txt [hunk 9]
```diff
@@ -20,3 +20,23 @@ You will face compatibility issues trying to run a different Teleport version wi
 If you want to run Teleport version {{.Values.teleportVersionOverride}},
 you should use `helm --version {{.Values.teleportVersionOverride}}` instead.
 {{- end }}
+
+
+{{- if contains "-gke." .Capabilities.KubeVersion.Version -}}
+{{- $groupName := (coalesce .Values.adminClusterRolebinding.name "cluster-admin") }}
```
Comment (webvictim) [inline examples/chart/teleport-kube-agent/templates/NOTES.txt:26]:
```suggestion
{{- $groupName := (coalesce .Values.adminClusterRoleBinding.name "cluster-admin") }}
```

File: examples/chart/teleport-kube-agent/templates/NOTES.txt [hunk 10]
```diff
@@ -20,3 +20,23 @@ You will face compatibility issues trying to run a different Teleport version wi
 If you want to run Teleport version {{.Values.teleportVersionOverride}},
 you should use `helm --version {{.Values.teleportVersionOverride}}` instead.
 {{- end }}
+
+
```
Comment (webvictim) [inline examples/chart/teleport-kube-agent/templates/NOTES.txt:24]:
```suggestion

```

**Review by webvictim — APPROVED** [review]
LGTM with one nit on `binding` -> `Binding` just to keep things consistent

File: examples/chart/teleport-kube-agent/templates/NOTES.txt [hunk 11]
```diff
@@ -20,3 +20,21 @@ You will face compatibility issues trying to run a different Teleport version wi
...
+{{- if contains "-gke." .Capabilities.KubeVersion.Version -}}
+{{- $groupName := (coalesce .Values.adminClusterRoleBinding.name "cluster-admin") }}
+
+WARNING: GKE Autopilot clusters forbid users from impersonating system-wide identities.
+This means you won't be able to use the `system:masters` Kubernetes Group in
+the Teleport Roles for GKE Autopilot clusters.
+
+Given that you installed Teleport on a GKE cluster, we recommend you use the
+Kubernetes Group `{{ $groupName }}` instead of `system:masters` in the Teleport Roles
+for GKE Autopilot clusters.
```
Comment (r0mant) [inline examples/chart/teleport-kube-agent/templates/NOTES.txt:32]: Can we also include a role snippet example below showing appropriate `kubernetes_groups: ["{{ $groupName }}"]` rule in a role for clarity?

**Review by r0mant — APPROVED** [review]

**Comment by hugoShaka** [comment]
Hey, sorry for the delay I was out yesterday.

I don't think it's a good idea to show this warning to all GKE users by default. From my experience, the vast majority of GKE users don't use autopilot. As a non-autopilot user this message is both non-actionable and misleading as it really sounds like I must do something. I understand that we want defaults that works on all clusters, but IMO this will make the experience with for all GKE users.

**Pull request merged**